table.cc

```diff
@@ -2008,6 +2008,10 @@ void table::set_schema(schema_ptr s) {
         _counter_cell_locks->set_schema(s);
     }
     _schema = std::move(s);
+    _config.enable_cache = ks.get_config().enable_cache && _schema->caching_options().enabled();
+    if (!_config.enable_cache) {
+        _cache.invalidate([] {}).get();
+    }
```

Comment:
Reads that are in-progress while this is happening can still populate the cache after invalidate() completes.

That's not a huge problem since the cache will still be coherent, and when we enable it again the data from those reads will be valid.

Maybe it's better to drop the invalidate() call. The data in the cache will age out naturally, and this allows ALTER TABLE to complete more quickly.

Reply:
I see no problem in removing this. Will do.